File: library/src/main/java/me/proxer/library/api/list/ListApi.java
package me.proxer.library.api.list;

import retrofit2.Retrofit;

/**
 * API for the List class.
 *
 * @author Desnoo.
 */
public final class ListApi {

    private final InternalApi internalApi;

    public ListApi(final Retrofit retrofit) {
        this.internalApi = retrofit.create(InternalApi.class);
    }

    /**
     * Returns the respective endpoint.
     */
    public MediaListEndpoint mediaList() {
        return new MediaListEndpoint(internalApi);
    }

    /**
     * Returns the respective endpoint.
     */
    public MediaSearchEndpoint mediaSearch() {
        return new MediaSearchEndpoint(internalApi);
    }

    /**
     * Returns the respective endpoint.
     */
    public TranslatorGroupListEndpoint translatorGroupList() {
        return new TranslatorGroupListEndpoint(internalApi);
    }

    /**
     * Returns the respective endpoint.
     */
    public IndustryListEndpoint industryList() {
        return new IndustryListEndpoint(internalApi);
    }

    /**
     * Returns the respective endpoint.
     */
    public TranslatorGroupProjectListEndpoint translatorGroupProjectList(final String translatorGroupId) {
        return new TranslatorGroupProjectListEndpoint(internalApi, translatorGroupId);
    }

    /**
     * Returns the respective endpoint.
     */
    public IndustryProjectListEndpoint industryProjectList(final String industryId) {
        return new IndustryProjectListEndpoint(internalApi, industryId);
    }

    /**
     * Returns the respective endpoint.
     */
    public TagListEndpoint tagList() {
        return new TagListEndpoint(internalApi);
    }
}

File: library/src/main/kotlin/me/proxer/library/api/list/ListApi.kt
package me.proxer.library.api.list

import retrofit2.Retrofit

/**
 * API for the List class.
 *
 * @author Desnoo.
 */
class ListApi internal constructor(retrofit: Retrofit) {

    private val internalApi = retrofit.create(InternalApi::class.java)

    /**
     * Returns the respective endpoint.
     */
    fun mediaList(): MediaListEndpoint {
        return MediaListEndpoint(internalApi)
    }

    /**
     * Returns the respective endpoint.
     */
    fun mediaSearch(): MediaSearchEndpoint {
        return MediaSearchEndpoint(internalApi)
    }

    /**
     * Returns the respective endpoint.
     */
    fun translatorGroupList(): TranslatorGroupListEndpoint {
        return TranslatorGroupListEndpoint(internalApi)
    }

    /**
     * Returns the respective endpoint.
     */
    fun industryList(): IndustryListEndpoint {
        return IndustryListEndpoint(internalApi)
    }

    /**
     * Returns the respective endpoint.
     */
    fun translatorGroupProjectList(translatorGroupId: String): TranslatorGroupProjectListEndpoint {
        return TranslatorGroupProjectListEndpoint(internalApi, translatorGroupId)
    }

    /**
     * Returns the respective endpoint.
     */
    fun industryProjectList(industryId: String): IndustryProjectListEndpoint {
        return IndustryProjectListEndpoint(internalApi, industryId)
    }

    /**
     * Returns the respective endpoint.
     */
    fun tagList(): TagListEndpoint {
        return TagListEndpoint(internalApi)
    }
}
